homeassistant/components/opentherm_gw.py

```diff
@@ -72,3 +77,86 @@
+    sensor_type_map = {
+        COMP_SENSOR: [
+            gw_vars.DATA_CONTROL_SETPOINT,
+            gw_vars.DATA_MASTER_MEMBERID,
+            gw_vars.DATA_SLAVE_MEMBERID,
+            gw_vars.DATA_SLAVE_OEM_FAULT,
+            gw_vars.DATA_COOLING_CONTROL,
+            gw_vars.DATA_CONTROL_SETPOINT_2,
+            gw_vars.DATA_ROOM_SETPOINT_OVRD,
+            gw_vars.DATA_SLAVE_MAX_RELATIVE_MOD,
+            gw_vars.DATA_SLAVE_MAX_CAPACITY,
+            gw_vars.DATA_SLAVE_MIN_MOD_LEVEL,
+            gw_vars.DATA_ROOM_SETPOINT,
+            gw_vars.DATA_REL_MOD_LEVEL,
+            gw_vars.DATA_CH_WATER_PRESS,
+            gw_vars.DATA_DHW_FLOW_RATE,
+            gw_vars.DATA_ROOM_SETPOINT_2,
+            gw_vars.DATA_ROOM_TEMP,
+            gw_vars.DATA_CH_WATER_TEMP,
+            gw_vars.DATA_DHW_TEMP,
+            gw_vars.DATA_OUTSIDE_TEMP,
+            gw_vars.DATA_RETURN_WATER_TEMP,
+            gw_vars.DATA_SOLAR_STORAGE_TEMP,
+            gw_vars.DATA_SOLAR_COLL_TEMP,
+            gw_vars.DATA_CH_WATER_TEMP_2,
+            gw_vars.DATA_DHW_TEMP_2,
+            gw_vars.DATA_EXHAUST_TEMP,
+            gw_vars.DATA_SLAVE_DHW_MAX_SETP,
+            gw_vars.DATA_SLAVE_DHW_MIN_SETP,
+            gw_vars.DATA_SLAVE_CH_MAX_SETP,
+            gw_vars.DATA_SLAVE_CH_MIN_SETP,
+            gw_vars.DATA_DHW_SETPOINT,
+            gw_vars.DATA_MAX_CH_SETPOINT,
+            gw_vars.DATA_OEM_DIAG,
+            gw_vars.DATA_TOTAL_BURNER_STARTS,
+            gw_vars.DATA_CH_PUMP_STARTS,
+            gw_vars.DATA_DHW_PUMP_STARTS,
+            gw_vars.DATA_DHW_BURNER_STARTS,
+            gw_vars.DATA_TOTAL_BURNER_HOURS,
+            gw_vars.DATA_CH_PUMP_HOURS,
+            gw_vars.DATA_DHW_PUMP_HOURS,
+            gw_vars.DATA_DHW_BURNER_HOURS,
+            gw_vars.DATA_MASTER_OT_VERSION,
+            gw_vars.DATA_SLAVE_OT_VERSION,
+            gw_vars.DATA_MASTER_PRODUCT_TYPE,
+            gw_vars.DATA_MASTER_PRODUCT_VERSION,
+            gw_vars.DATA_SLAVE_PRODUCT_TYPE,
+            gw_vars.DATA_SLAVE_PRODUCT_VERSION,
+            gw_vars.OTGW_MODE,
+            gw_vars.OTGW_DHW_OVRD,
+            gw_vars.OTGW_ABOUT,
+            gw_vars.OTGW_BUILD,
+            gw_vars.OTGW_CLOCKMHZ,
+            gw_vars.OTGW_LED_A,
+            gw_vars.OTGW_LED_B,
+            gw_vars.OTGW_LED_C,
+            gw_vars.OTGW_LED_D,
+            gw_vars.OTGW_LED_E,
+            gw_vars.OTGW_LED_F,
+            gw_vars.OTGW_GPIO_A,
+            gw_vars.OTGW_GPIO_B,
+            gw_vars.OTGW_SB_TEMP,
+            gw_vars.OTGW_SETP_OVRD_MODE,
+            gw_vars.OTGW_SMART_PWR,
+            gw_vars.OTGW_THRM_DETECT,
+            gw_vars.OTGW_VREF,
+        ]
+    }
+    sensors = []
+    for var in monitored_vars:
+        if var in sensor_type_map[COMP_SENSOR]:
+            sensors.append(var)
+        else:
+            _LOGGER.error("Monitored variable not supported: %s", var)
+    if sensors:
+        hass.async_create_task(
```

Comment:
We can just await the coroutine here.

Reply:
This was changed in 03be3ee, but I am considering to change it back, either now or in the next PR. The rationale behind it is that I want to add the binary sensors from here as well and I figured it would be more efficient to schedule both tasks at once rather than awaiting them.